### PR TITLE
Highcharts.objectEach()

### LINKED ISSUE #5482: GeoJSON Transform object iteration error
Hi, apologies I skipped the issue template because it didnt make any sense for this issue.

I was trying to understand the longitude-latitude transformation for geojson where you have used proj4js and I found this glitch

I might have missed something but this for block
https://github.com/highcharts/highcharts/blob/25e4b0a7b2298faa9fc9c0b23eb95e46d4a9bd6e/lib/modules/map.src.js#L1337

should have been

``` js
for (transformKey in mapTransforms) {
       var transform = mapTransforms[transformKey];
    if (mapTransforms.hasOwnProperty(transformKey) && transform.rotation) {
          transform.cosAngle = Math.cos(transform.rotation);
          transform.sinAngle = Math.sin(transform.rotation);
       }
}
```

### PR DESCRIPTION
**Pull request by larsac07**
Acts just like the `Highcharts.each()`, except it iterates over object key-value-pairs. Saves a lot of unnecessary index variables and ensures that `hasOwnProperty()` is checked (was lacking in many of the refactored `for in`-loops). Also seems to fix #5482

### CODE REVIEW

File: js/parts/Utilities.js [hunk 1]
```diff
@@ -1544,6 +1544,26 @@ H.each = function (arr, fn, ctx) { // modern browsers
...
+ * @param  {Object}   obj - The object to iterate over.
+ * @param  {Function} fn  - The iterator callback. It passes three arguments:
+ * * value - The property value.
+ * * key - The property key.
+ * * obj - The object that objectEach is being applied to.
+ * @param  {Object}   ctx The context
+ */
+H.objectEach = function (obj, fn, ctx) {
+	var key;
+	for (key in obj) {
```
Comment (oysteinmoseng) [inline js/parts/Utilities.js]: Would use `for (var in obj)` to save 4 precious bytes

**Comment by oysteinmoseng** [comment]
+1 for jon's suggestion

**Comment by jon-a-nygaard** [comment]
Also think that it is better in Treemap.js to replace the `eachObject` function completely with the new `objectEach` at [L28-L33](https://github.com/highcharts/highcharts/blob/7c883198706422c086ff58e9cfb6d678f68e8faf/js/modules/treemap.src.js#L28-L33) and [L125-L132](https://github.com/highcharts/highcharts/blob/7c883198706422c086ff58e9cfb6d678f68e8faf/js/modules/treemap.src.js#L125-L132).
That would also mean that the callback in objectEach will have to get the full object as the third parameter, but it is a common pattern which I believe the function should support.

**Comment by jon-a-nygaard** [comment]
This is nitpicking, but I think `stackItem` is more correct naming in stead of `stackX` for the values in a stack: [Series.js#L62-L66](https://github.com/highcharts/highcharts/blob/7c883198706422c086ff58e9cfb6d678f68e8faf/js/parts/AreaSeries.js#L62-L66)

File: js/parts/Utilities.js (hunk 1 above)
Comment (TorsteinHonsi) [inline js/parts/Utilities.js]: Actually not - Google Closure Compiler takes care of this. We just need to decide what we find more pleasing to the eye.

**Pull request merged**